contracts/adapters/HegicAdapter.sol

```diff
@@ -352,19 +352,55 @@ contract HegicAdapter is IProtocolAdapter {
     }
 
     /**
-     * @notice Function to bulk claim rHEGIC2 rewards from liquidity utilization
+     * @notice Function to get rHEGIC2 rewards claimable from liquidity utilization
+     * @param rewardsAddress is the address of the rewards contract (either for eth or wbtc)
+     * @param optionIDs is an array of all the option ids we want to claim rewards for
+     */
+    function rewardsClaimable(address rewardsAddress, uint256[] calldata optionIDs)
+        external
+        view
+        returns (uint256 rewardsAmount)
+    {
+        IHegicRewards rewardsContract = IHegicRewards(rewardsAddress);
+
+        uint256 i = 0;
+
+        while (i < optionIDs.length && optionIDs[i] > 0) {
+          (, , , uint _amount, , uint _premium, , ) = rewardsContract.hegicOptions().options(optionIDs[i]);
+          if(!rewardsContract.rewardedOptions(optionIDs[i])){
+            rewardsAmount = rewardsAmount.add(_amount.div(100).add(_premium)
+            .mul(rewardsContract.rewardsRate())
+            .div(1e8));
+          }
+          i += 1;
+        }
+    }
+
+    /**
+     * @notice Function to get rHEGIC2 rewards claimable from liquidity utilization
      * @param rewardsAddress is the address of the rewards contract (either for eth or wbtc)
      * @param optionIDs is an array of all the option ids we want to claim rewards for
      */
     function claimRewards(address rewardsAddress, uint256[] calldata optionIDs)
         external
+        returns (uint256 rewardsAmount)
     {
         IHegicRewards rewardsContract = IHegicRewards(rewardsAddress);
+
         uint256 i = 0;
-        while (i < optionIDs.length) {
-          rewardsContract.getReward(optionIDs[i]);
+
+        uint256 balanceBefore = rewardsContract.hegic().balanceOf(address(this));
```

Comment:
Similar to [this comment](https://github.com/ribbon-finance/structured-products/pull/59/files#r582642378), we can store the result of `rewardsContract.hegic()` in a variable to save 200 gas.

```
IERC20 hegicToken = rewardsContract.hegic();
uint256 balanceBefore = hegicToken.balanceOf(address(this));
// ...
uint256 balanceAfter = hegicToken.balanceOf(address(this));
```